app/models/concerns/commentable.rb

```diff
@@ -4,4 +4,10 @@ module Commentable
   included do
     has_many :comments, as: :commentable, dependent: :destroy
   end
+
+  def commentable_activities
+    Activity.where(trackable_type: self.class.to_s, trackable_id: self.id).or(
+      Activity.where(trackable_type: 'Comment', trackable_id: [self.comments.map(&:id)])
```

Comment:
I'd use `self.comments.pluck(:id)`

Reply:
I'd go one better and use `comment_ids`  ;)

http://guides.rubyonrails.org/association_basics.html#methods-added-by-has-many